openhtf/io/frontend/src/app/dashboard.service.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016 Google Inc.All Rights Reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+import {Injectable} from 'angular2/core';
+import {Http, Response} from 'angular2/http';
+
+import 'rxjs/Rx';
+
+
+/** Provider of Dashboard updates from the OpenHTF frontend server. **/
+@Injectable()
+export class DashboardService {
+  POLLING_INTERVAL_MS: number = 3000;
+  URL: string = '/raw/dashboard';
+
+  listings: Listing[];
+  dashMap: {};
+  server: any;
+  polling_job: number;
+
+  /**
+   * Create a DashboardService.
+   */
+  constructor(private http: Http) {
+    this.listings = []; // Because ngFor only takes lists.
+    this.dashMap = {}; // Because we want persistant records.
```

Comment:
Good eye. Eagerly awaiting the rest of your feedback. :stuck_out_tongue_winking_eye: